theano/tensor/opt.py

```diff
@@ -2204,7 +2204,7 @@ def local_cast_cast(node):
 
     when those contrain:
     dtype1 == dtype2
-    TODO: the base dtype is the same (int, uint, float, complex)
+    OR the base dtype is the same (int, uint, float, complex)
           and the first cast cause an upcast.
 
     """
```

Comment:
Is this handled by this local_cast_cast function? I think it's eliminated at an earlier stage of the optimization. This function doesn't seem to be called at all when something like "x{fvector}.cast('float32') -> x" is done. Although I can't trace where exactly.

Also, a related question : is the type1 == type2 comparison also redundant? I think it's eliminated at the same previous stage, ie, x.cast(dtype1).cast(dtype1) gets reduced to x.cast(dtype1) without the aid of the local_cast_cast function.